Gems/LmbrCentral/Code/Source/Builders/LuaBuilder/LuaBuilderWorker.cpp

```diff
@@ -57,17 +74,11 @@ namespace LuaBuilder
                         asset.SetAutoLoadBehavior(AZ::Data::AssetLoadBehavior::PreLoad);
                         assets.push_back(asset);
                     }
-                    else if(isLuaDependency)
-                    {
-                        AZ_Error("LuaBuilder", false, "Did not find dependency %s referenced by script.", dependency.m_dependencyPath.c_str());
-                    }
                     else
                     {
-                        AZ_Error("LuaBuilder", false, "%s referenced by script does not appear to be a lua file.\n"
-                            "References to assets should be handled using Property slots and AssetIds to ensure proper dependency tracking.\n"
-                            "This file will not be tracked as a dependency.",
-                            dependency.m_dependencyPath.c_str());
+                        AZ_Error("LuaBuilder", false, "Invalid asset %s referenced by script.", dependency.m_dependencyPath.c_str());
                     }
+                    // carbonated end
```

Comment:
You did not change the later code where every job that processes assets in the list gets a lua-related string:
            for (auto& dependentAsset : dependentAssets)
            {
                AssetBuilderSDK::JobDependency jobDependency;
                jobDependency.m_sourceFile.m_sourceFileDependencyUUID = dependentAsset.GetId().m_guid;
                jobDependency.m_jobKey = "Lua Compile";

Reply:
Looks like jobKey for dependency is skipped always, when dependency jobs are analyzed (see AssetProcessorManager::UpdateSourceFileDependenciesDatabase for example) and later AP sets an appropriate jobKey for any resource type from dependency or remove duplicated dependencies.
 
I can't catch no one case in AP at the page "Jobs" in column "Job Key"  when "Lua Compile" appears for any kind of assets different with LUA. "Job Key" column are always displayed properly. Need further investigation.

In any case, what is the Job Key:
```
            //! Job Key is arbitrarily defined by the builder.  Used to differentiate between different jobs emitted for the same input file, for the same platform, for the same builder.
            //! for example, you might want to split a particularly complicated and time consuming job into multiple sub-jobs.  In which case they'd all have the same input file,
            //! the same platform, the same builder UUID (since its the UUID of the builder itself)
            //! but would have different job keys.
```